diff --git a/arch/arm/dts/zynqmp-zcu102-rev1.1.dts b/arch/arm/dts/zynqmp-zcu102-rev1.1.dts
index b6798394..11658587 100644
--- a/arch/arm/dts/zynqmp-zcu102-rev1.1.dts
+++ b/arch/arm/dts/zynqmp-zcu102-rev1.1.dts
@@ -13,3 +13,7 @@
 	model = "ZynqMP ZCU102 Rev1.1";
 	compatible = "xlnx,zynqmp-zcu102-rev1.1", "xlnx,zynqmp-zcu102", "xlnx,zynqmp";
 };
+
+&sdhci1 {
+    /delete-property/ no-1-8-v;
+};
\ No newline at end of file
